### PR TITLE
compendium and RNA-seq Sample compendium

### PR DESCRIPTION
**Pull request by davidsmejia**
## Issue Number

#758 

## Purpose/Implementation Notes

- Implements the new compendium and RNA-seq Sample Compendia 
- ThemeContext.js
- DownloadCompendia takes a filter option which allows for arbitrary filtering for all of your future compendia needs
- Header now has a DropDown links component which takes an array of objects with keys ```title``` to display for the link and ```location``` which is a ```react-router-dom``` [object or string](https://reacttraining.com/react-router/web/api/Link)

ThemeContext.js:
In order to solve the problem with the various headers on the site overriding the new dropdown navigation elements I have implemented a theme context that allows pages to tell other components which color they would like them to be... or handle a specific color request. It has a simple interface which uses React Hooks under the hood.
ie ```import {useTheme} from './common/ThemeContext.js';```
and then if you want to set a theme:
```useTheme('light')```
and if you want to handle themes:
```const [theme] = useTheme()```
then ``` ... if (theme.thePropertyIcareAbout === 'light') makeThingLight()```
so any component can decide how if and how it should handle that... 

Just a thought:
Also.. ThemeContext.js should probably go to ```src/common/hooks.js``` and use a config file from ```src/common/themes.js``` instead

## Types of changes

* [ ] Bugfix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Functional tests

Yarn tests, can download 

Note: large files seem to take a long time and there is an issue with Xorigin cookies that is probably coming from AWS

## Checklist

* [x] Lint and unit tests pass locally with my changes
* [ ] I have added tests that prove my fix is effective or that my feature works
* [ ] I have added necessary documentation (if appropriate)
* [ ] Any dependent changes have been merged and published in downstream modules

## Screenshots

Screen Shot 2019-10-09 at 7 07 13 PM

### CODE REVIEW

File: src/pages/Compendia/index.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import classnames from 'classnames';
+import { Link, withRouter, Redirect } from 'react-router-dom';
+import './Compendia.scss';
+import PointsBackground from './PointsBackground';
+import NormalizedCompendia from './NormalizedCompendia';
+import RNASeqSampleCompendia from './RNASeqSampleCompendia';
+import EmailSection from '../Main/EmailSection';
+import { useTheme } from '../../common/ThemeContext';
+
+const Compendia = function Compendia({ location, replace }, ...props) {
```
Comment (arielsvn) [inline src/pages/Compendia/index.js]: What is `...props` here for? Also, you could probably remove the variable declaration

File: src/pages/Main/Main.scss [hunk 2]
```diff
@@ -90,14 +86,12 @@
...
+      box-shadow: 0 3px 20px 0 rgba(0, 0, 0, 0.2);
       overflow: hidden;
 
-      @include tablet {
-        padding: 10 * 8px 0 0;
-      }
-
       @include desktop() {
         overflow: visible;
+        min-height: 487px;
```
Comment (arielsvn) [inline src/pages/Main/Main.scss]: Can we use a multiple of `8px` here? `480` maybe?

File: src/components/Header/index.js [hunk 3]
```diff
@@ -13,23 +13,17 @@ import SideMenu from '../SideMenu';
...
 import githubCorner from './github-corner.svg';
+import { push } from '../../state/routerActions';
 
-/**
- * In general this is a bad approach, where the header component knows about other pages.
- * But in the future we'll unify the header styles and have a single color. So it doesn't make
- * much sense to invest a lot of time improving this.
- */
-function shouldInvertColors(pathname) {
-  return ['/', '/about', '/species-compendia'].includes(pathname);
```
Comment (arielsvn) [inline src/components/Header/index.js]: 🔥 🔥 loving the `useTheme` approach for removing this atrocity!

File: src/common/ThemeContext.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    if (themeName) setTheme(themeName);
+    return () => {
+      if (themeName) setTheme('default');
+    };
+  }, [themeName, setTheme]);
+
+  return [theme, setTheme];
+};
+
+export { themes, ThemeContext, ThemeProvider, useTheme };
```
Comment (arielsvn) [inline src/common/ThemeContext.js]: What do you think of making `useTheme` private here and exporting two more specific hooks: `useThemeInverted()` and `useThemeLight()`?

File: src/common/ThemeContext.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+const themes = {
+  default: {
+    header: 'default',
+  },
+  inverted: {
+    header: 'inverted',
+  },
+  light: {
+    header: 'light',
+  },
```
Comment (arielsvn) [inline src/common/ThemeContext.js]: Why have the specific value for `header` here? Should the `Header` component know how to display itself depending just on the active theme?

File: src/components/Header/index.js [hunk 6]
```diff
@@ -125,6 +136,80 @@ const HeaderLink = ({ to, onClick, children, location, activePath = [] }) => {
   );
 };
 
+let HeaderDropDownLink = ({
+  to = [],
+  onClick,
+  children,
+  location,
+  activePath = [],
```
Comment (arielsvn) [inline src/components/Header/index.js]: `activePath` is needed?

File: src/pages/Compendia/DownloadCompendia.js [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+  state => ({
+    token: state.token,
+  }),
+  { createToken, push }
+)(DownloadCompendia);
+export default DownloadCompendia;
+
+const filterForLatestCompendia = (data, organism = false) => {
+  data.forEach(c => {
+    c.organism_name = c.result.annotations[0].data.organism_name;
```
Comment (arielsvn) [inline src/pages/Compendia/DownloadCompendia.js]: Is there a way to achieve the same here without [mutating the parameter](https://github.com/airbnb/javascript#functions--mutate-params)?

File: src/components/Header/index.js [hunk 8]
```diff
@@ -125,6 +136,80 @@ const HeaderLink = ({ to, onClick, children, location, activePath = [] }) => {
...
+          </ul>
+        </li>
+      </ul>
+    </li>
+  );
+};
+
+HeaderDropDownLink = connect(
+  null,
+  { push }
```
[the remark this answers is not in the document]
Comment (davidsmejia) [reply src/components/Header/index.js]: i think we can remove this 👍

File: src/pages/Compendia/DownloadCompendia.js [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+    c => c.organism_name === organism.organism_name
+  );
+  if (downloadableFile && downloadableFile.download_url) {
+    return downloadableFile.download_url;
+  }
+  // throw error? or alert slack
+  return '';
+}
+
+const getPrettyFileSize = size => {
```
[the remark this answers is not in the document]
Comment (davidsmejia) [reply src/pages/Compendia/DownloadCompendia.js]: removed

File: src/pages/Compendia/index.js (hunk 1 above)
Comment (davidsmejia) [reply src/pages/Compendia/index.js]: removed

File: src/components/Header/index.js (hunk 6 above)
Comment (davidsmejia) [reply src/components/Header/index.js]: yeah i just forgot to implement it :) good catch!

File: src/pages/Compendia/DownloadCompendia.js [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+
+  const organismNames = uniq(data.map(c => c.organism_name));
+  const filtered = [];
+
+  for (const organismName of organismNames) {
+    const organismCompendium = data
+      .filter(c => c.organism_name === organismName)
+      .sort((x, y) => y.compendia_version - x.compendia_version);
+    filtered.push(organismCompendium[0]);
+  }
```
[the remark this answers is not in the document]
Comment (davidsmejia) [reply src/pages/Compendia/DownloadCompendia.js]: will do

File: src/pages/Main/Main.scss (hunk 2 above)
Comment (davidsmejia) [reply src/pages/Main/Main.scss]: yup

**Comment by cgreene** [comment]
Design discussed in #776

File: src/common/ThemeContext.js (hunk 5 above)
Comment (davidsmejia) [reply src/common/ThemeContext.js]: The idea here is that each theme can be composed of different colors for each component.

File: src/common/ThemeContext.js (hunk 4 above)
Comment (davidsmejia) [reply src/common/ThemeContext.js]: I've updated this to use ```useTheme(themes.light)``` you can ```import {themes, useTheme } from 'ThemeContext' ```

File: src/api/compendia.js [hunk 11]
```diff
@@ -0,0 +1,33 @@
+import { Ajax } from '../common/helpers';
+
+export async function fetchCompendiaData(additionalFilters = {}, token) {
+  try {
+    const filters = {
+      ...{
```
Comment (arielsvn) [inline src/api/compendia.js]: You can write these properties directly instead of spreading them?

File: src/api/compendia.js [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+        ordering: 'primary_organism__name',
+        limit: 1000,
+      },
+      ...additionalFilters,
+    };
+
+    const values = Object.entries(filters).map(([key, value]) => {
+      return `${key}=${value}`;
+    });
+    const endpoint = `/v1/compendia/?${values.join('&')}`;
```
Comment (arielsvn) [inline src/api/compendia.js]: You can also pass `filters` as the second parameter of `Ajax.get`, and it'll do something like this.

File: src/api/compendia.js [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+
+    const response = await Ajax.get(endpoint, null, tokenObject);
+
+    return response.results;
+  } catch (e) {
+    return [];
+  }
+}
+
+export async function fetchCompendium(token, selected) {
```
Comment (arielsvn) [inline src/api/compendia.js]: what if we pass the compendium `id` as a parameter, instead of `selected`?

File: .eslintrc.js [hunk 14]
```diff
@@ -62,6 +62,9 @@ module.exports = {
     'prefer-destructuring': 'warn',
     'no-param-reassign': 'warn',
 
+    // this should be fixed in header/headerA
+    'jsx-a11y/no-noninteractive-element-interactions': 'warn',
```
Comment (arielsvn) [inline .eslintrc.js]: I think we talked about this, but I can't remember well... Did this caused issues?

File: src/pages/Compendia/DownloadCompendia.js [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+  title,
+  filter,
+  children,
+}) => {
+  const agreedToTerms = !!token;
+  const [selected, setSelected] = React.useState(null);
+  const [agree, setAgree] = React.useState(agreedToTerms);
+  const { data, isLoading } = useLoader(() => fetchCompendiaData(filter));
+
+  if (isLoading)
```
Comment (arielsvn) [inline src/pages/Compendia/DownloadCompendia.js]: Can we use braces here? [ref](https://github.com/airbnb/javascript#blocks--braces)

File: src/pages/Compendia/DownloadCompendia.js [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+    return (
+      <div className="download-compendia">
+        <p className="download-compendia__title">Fetching {title}</p>
+        <Spinner />
+      </div>
+    );
+  if (!isLoading && data.length === 0)
```
Comment (arielsvn) [inline src/pages/Compendia/DownloadCompendia.js]: `!isLoading` is always true here (from lgtm)

Comment (davidsmejia) [reply src/pages/Compendia/DownloadCompendia.js]: 👍 changing to an else if clause

**Pull request merged**